### ISSUE TITLE
Should `MODEL.ROI_HEADS.NUM_CLASSES` include background? 

### ISSUE THREAD

**Issue by FraPochetti** [facebookresearch/detectron2#2367]
Hello everybody,
first of all, thanks a lot for building this awesome library!

I have checked the docs and could not find a clear answer to my question, so asking here.
I am currently using a `mask_rcnn` model and I am not sure whether `MODEL.ROI_HEADS.NUM_CLASSES` should include _background_ or not (e.g. if my dataset includes only `dogs`, NUM_CLASSES should be `2`).

This is not specified in the relevant [class docstring](https://github.com/facebookresearch/detectron2/blob/9bfa212f68daf47b42e9bb1314652cfce5882d13/detectron2/modeling/roi_heads/mask_head.py#L220) and, given `torchvision` [makes it explicit](https://github.com/pytorch/vision/blob/4d3a309379f2df748922a610c5dd648ae7248bae/torchvision/models/detection/mask_rcnn.py#L315), I am a bit confused.

Can you please provide some guidance?
Thanks

**Comment by huangfeng95** [comment]
I have the same doubt

**Comment by ppwwyyxx** [comment]
The config reference https://detectron2.readthedocs.io/modules/config.html#config-references has answered this.

We will add it to the class docstring as well.

**Comment by FraPochetti** [comment]
I am more than happy to open a PR for this!

@ppwwyyxx thanks a lot for the prompt reply.
I just want to make sure I get this right.

_"Number of foreground classes"_ means that it should include `background`, so in my example above "if my dataset includes only `dogs`, NUM_CLASSES should be `2`".
Is my understanding correct?

**Comment by ppwwyyxx** [comment]
> "Number of foreground classes" means that it should include background

I'm curious why this is understood this way? foreground and background are two opposite word, so I would interpret it that "Foreground classes" do not include "background"

**Comment by FraPochetti** [comment]
You are right. 
The reason is that I misinterpreted `foreground` and `background` as synonyms.
Just bad English on my side I guess. My apologies.

I think it would probably be an idea to be more specific in the docs, explicitly mentioning  `number of foreground classes, e.g. not including background`.
Some sort of "explain it to me like I am 5" to avoid any other bad English speakers to get confused in the future.

Thanks a lot for the reply!
Do you want me to open a PR about this?

**Comment by binhere** [comment]
> The config reference https://detectron2.readthedocs.io/modules/config.html#config-references has answered this: "Number of foreground classes"
> 
> We will add it to the class docstring as well.

Hi @ppwwyyxx 

Thank you for your instruction, It is very helpful; however, this document about dataset https://detectron2.readthedocs.io/en/latest/tutorials/datasets.html#standard-dataset-dicts also says: 

"category_id (int, required): an integer in the range [0, num_categories-1] representing the category label. The value num_categories is reserved to represent the “background” category, if applicable."

Actually, I am confused about that statement "The value num_categories is reserved to represent the “background” category, if applicable." 
**How do you understand this ?**

In my case, when I create a dataset on Roboflow and export it as COCO segmentation format, my dataset (has only 1 class "shelf" ) always includes class 0 for background. Here is my json file about categories field:

categories": [
    {
      "id": 0,
      "name": "0",
      "supercategory": "none"
    },
    {
      "id": 1,
      "name": "shelf",
      "supercategory": "0"
    }
  ]

**I am now struggling with these things:**

`MetadataCatalog.get("train").thing_classes` should include background? for example, `["0", "shelf"]` ?
or 
how can I adjust my json format to be compatible with detectron2 expects ? 

However, follow the explicit document that you refered, I think my following code is correct:
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1 # for foreground class "shelf" only
